### PR TITLE
New As a user, I would like to be able to request money from someon, so I can get paid and remind a forgetful funder.

### PR DESCRIPTION
**Pull request by zoek1**
##### Refers/Fixes
As a user, I would like to be able to request money from someon, so I can get paid and remind a forgetful funder. #6193

##### Testing

**Demo**: https://www.dropbox.com/s/3jmejmc052hyjfy/2020-03-24%2011-56-28.flv?dl=0

image

**Email template**
DeepinScreenshot_20200330010618

### CODE REVIEW

**Comment by owocki** [comment]
thanks; i'm very excited about this. just put it on the @gitcoinco/engineers PR review board

**Comment by danlipert** [comment]
@zoek1 Does this work with DAI? it seems that it only works for Ether right now

File: app/dashboard/tip_views.py [hunk 1]
```diff
@@ -60,6 +61,45 @@ def send_tip(request):
...
+def request_money(request):
+    """"""
+    if request.method == 'POST':
+        username = request.POST.get('username', '').strip('@')
+        token_name = request.POST.get('tokenName')
+        amount = request.POST.get('amount')
+        comments = request.POST.get('comments')
+        network = request.POST.get('network')
+        address = request.POST.get('address')
+        token_address = request.POST.get('tokenAddress')
```
Comment (danlipert) [inline app/dashboard/tip_views.py:73]: Where is this used?

Comment (zoek1) [reply app/dashboard/tip_views.py]: Sorry, i forgot delete this unused variable

**Comment by zoek1** [comment]
Updated @da

> @zoek1 Does this work with DAI? it seems that it only works for Ether right now

Yes, it works with DAI even with other networks like SIA, or ETC. The request payment view only requires the token name, so the tip view can retrieve the token address.

**Comment by zoek1** [comment]
@thelostone-mc could you check this PR please? :smiley:

File: app/assets/onepager/js/request.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    method: 'POST',
+    body: formData
+  }).then(function(json) {
+    _alert('The funder has been notified', 'success');
+    success_callback();
+  }).catch(function(error) {
+    _alert('Something goes wrong, try later.', 'error');
+    failure_callback();
+  });
+}
```
Comment (thelostone-mc) [inline app/assets/onepager/js/request.js:98]: ^ could we update all `var` to `const`/`let`?

File: app/dashboard/templates/onepager/send2.html [hunk 3]
```diff
@@ -80,18 +80,24 @@ <h1>{% trans "Send Tip." %}</h1>
...
-      </select>
-      <input name="amount" type="text" placeholder="1.1" id="amount" value="" style="width: 170px; display: inline;">
+        {% if fund_request.network == 'ETH' %}
+        <select id="token" style="width:100px; margin-bottom: 10px; display: inline;" data-token="{{ fund_request.token_name }}">
+          <option value="0x0">ETH</option>
+        </select>
+        {% else %}
+        <select style="width:100px; margin-bottom: 10px; display: inline;">
+            <option value="{{ fund_request.network }}">{{ fund_request.network }}</option>
+        </select>
```
Comment (octavioamu) [inline app/dashboard/templates/onepager/send2.html:90]: Hey @zoek1 just to let you know here was missing the id="token" so the token select wasn't getting fill.

Comment (zoek1) [reply app/dashboard/templates/onepager/send2.html:90]: Sorry @octavioamu, I'm not sure why the id was missing. I really tested :anguished:

**Pull request merged**